docs/source/dev/multimodal/adding_multimodal_plugin.rst

```diff
@@ -0,0 +1,17 @@
+.. _adding_multimodal_plugin:
+
+Adding a Multimodal Plugin
+==========================
+
+This document teaches you how to add a new modality to vLLM.
+
+Each modality in vLLM is represented by a :class:`~vllm.multimodal.MultiModalPlugin` and registered to :data:`~vllm.multimodal.MULTIMODAL_REGISTRY`.
+For vLLM to recognize a new modality type, you have to create a new plugin and then pass it to :meth:`~vllm.multimodal.MultiModalRegistry.register_plugin`.
+
+The remainder of this document details how to define custom :class:`~vllm.multimodal.MultiModalPlugin` s.
+
+.. note::
+  This article is a work in progress.
+
+..
```

Comment:
Perhaps we could take image as an example to show how to add a new modality?

Reply:
I thought you already had a plan for adding a tutorial (judging from the comment) so I haven't written up anything yet. Perhaps we can do that in another PR?

Reply:
Yep we can do that!